torchvision/transforms/functional.py

```diff
@@ -491,53 +491,76 @@ def hflip(img: Tensor) -> Tensor:
+    # RuntimeError: Expected type hint for result of tolist()
+    return [float(i.item()) for i in res[:, 0]]
 
 
-def perspective(img, startpoints, endpoints, interpolation=Image.BICUBIC, fill=None):
-    """Perform perspective transform of the given PIL Image.
+def perspective(
+        img: Tensor,
+        startpoints: List[List[int]],
+        endpoints: List[List[int]],
+        interpolation: int = 3,
+        fill: Optional[int] = None
+) -> Tensor:
+    """Perform perspective transform of the given image.
+    The image can be a PIL Image or a Tensor, in which case it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading dimensions.
 
     Args:
-        img (PIL Image): Image to be transformed.
-        startpoints: List containing [top-left, top-right, bottom-right, bottom-left] of the original image
-        endpoints: List containing [top-left, top-right, bottom-right, bottom-left] of the transformed image
-        interpolation: Default- Image.BICUBIC
+        img (PIL Image or Tensor): Image to be transformed.
+        startpoints (list of list of ints): List containing four lists of two integers corresponding to four corners
+            ``[top-left, top-right, bottom-right, bottom-left]`` of the original image.
+        endpoints (list of list of ints): List containing four lists of two integers corresponding to four corners
+            ``[top-left, top-right, bottom-right, bottom-left]`` of the transformed image.
+        interpolation (int): Interpolation type. If input is Tensor, only ``PIL.Image.NEAREST`` and
+            ``PIL.Image.BILINEAR`` are supported. Default, ``PIL.Image.BICUBIC`` for PIL images and
+            ``PIL.Image.BILINEAR`` for Tensors.
         fill (n-tuple or int or float): Pixel fill value for area outside the rotated
             image. If int or float, the value is used for all bands respectively.
-            This option is only available for ``pillow>=5.0.0``.
+            This option is only available for ``pillow>=5.0.0``. This option is not supported for Tensor
+            input. Fill value for the area outside the transform in the output image is always 0.
 
     Returns:
-        PIL Image:  Perspectively transformed Image.
+        PIL Image or Tensor: transformed Image.
     """
 
-    if not F_pil._is_pil_image(img):
-        raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
+    coeffs = _get_perspective_coeffs(startpoints, endpoints)
 
-    opts = _parse_fill(fill, img, '5.0.0')
+    if not isinstance(img, torch.Tensor):
+        return F_pil.perspective(img, coeffs, interpolation=interpolation, fill=fill)
 
-    coeffs = _get_perspective_coeffs(startpoints, endpoints)
-    return img.transform(img.size, Image.PERSPECTIVE, coeffs, interpolation, **opts)
+    if interpolation == Image.BICUBIC:
+        # bicubic is not supported by pytorch
+        # set to bilinear interpolation
+        interpolation = 2
```

Comment:
I'm not sure that silently changing the value requested by the user is a good thing to do here.

It is unfortunate that bicubic is not supported in PyTorch yet, but I think we should try to be consistent here (the user requested bicubic and we silently compute bilinear).

What about one of the following:
1. change the default value for the whole function to be `BILINEAR`, and raise an error if BICUBIC is passed by the user, and add a `[BC-breaking]` in the tile of the PR
2. raise a warning before switching the default value here to `bilinear`

I think I would prefer option 1 here, as option 2 would by default raise warnings to the user.

Reply:
I agree, that silent changing the value is not good at all ! It was a full-moon or something :)
I'll remove this part of code. Error is raised inside `F_t.affine` if resample is not support: 
https://github.com/pytorch/vision/pull/2558/files#diff-77c635250f0215f979b2650c383ecbecL637